build_tools/benchmarks/benchmark_helper.py

```diff
@@ -141,12 +145,79 @@ def _dump_cmds_handler(e2e_test_artifacts_dir: pathlib.Path,
   print(*lines, sep="\n")
 
 
+# Represents a benchmark results file with the data already loaded from a JSON file.
+class JSONBackedBenchmarkData:
+
+  def __init__(self, source_filepath: pathlib.PurePath, data: Dict):
+    if not isinstance(data, dict):
+      raise ValueError(
+          "\"{}\" seems not to be a valid benchmark-results-file (No JSON struct as root element)."
+          .format(source_filepath))
```

Comment:
Use alternative quotes rather than escaping. Also, this can be an f-string. Suggestion as an example, but same throughout here

```suggestion
          f"'{source_filepath}' seems not to be a valid benchmark-results-file (No JSON struct as root element).")
```

Reply:
Changed